scipy/sparse/linalg/isolve/tfqmr.py

```diff
@@ -0,0 +1,182 @@
+import warnings
+import numpy as np
+from .utils import make_system
+
+
+__all__ = ['tfqmr']
+
+
+def tfqmr(A, b, x0=None, tol=1e-5, maxiter=None, M=None,
+          callback=None, atol=None, view=False):
+    """
+    Use Transpose-Free Quasi-Minimal Residual iteration to solve ``Ax = b``.
+
+    Parameters
+    ----------
+    A : {sparse matrix, dense matrix, LinearOperator}
+        The real or complex N-by-N matrix of the linear system.
+        Alternatively, `A` can be a linear operator which can
+        produce ``Ax`` using, e.g.,
+        `scipy.sparse.linalg.LinearOperator`.
+    b : {array, matrix}
```

Comment:
Is `matrix` support needed? `np.matrix` is effectively deprecated, so I'd suggest leaving it out completely unless there's a good reason not to do so.

Reply:
`array` -> `ndarray`

Reply:
or maybe with "matrix" a sparse matrix was meant - then better to be more explicit